### PR TITLE
chore: add uds-runtime as an optional component in core

### PR DESCRIPTION
**Pull request by mjnagel**
## Description

Adds UDS Runtime as an optional component in core. Denoted as a chore since this is optional and we would like to cut a patch release including this for fast consumption.

## Related Issue

Closes https://github.com/defenseunicorns/uds-runtime/issues/346

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Comment by TristanHoladay** [comment]
LGTM

**Review by UncleGedd — DISMISSED** [review]
🚀

File: src/runtime/zarf.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+components:
+  - name: uds-runtime
+    required: false
+    images:
+      - ghcr.io/defenseunicorns/uds-runtime:0.4.0
+    charts:
+      - name: uds-runtime
+        namespace: uds-runtime
+        version: "v0.4.0"
```
Comment (rjferguson21) [inline src/runtime/zarf.yaml:15]: Should this not be 0.1.0? Aligned with https://github.com/defenseunicorns/uds-runtime/blob/main/chart/Chart.yaml#L18 ?

Comment (mjnagel) [reply src/runtime/zarf.yaml:15]: This is actually the git tag/version for the repo. Slightly confusing, but makes sure that we are pulling the chart for the same image. Might be good for runtime to synchronize the version of the chart as well - cc @UncleGedd ?

I do believe this works as is, even if not the most obvious - it does deploy in CI and locally as expected.

File: src/runtime/zarf.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+kind: ZarfPackageConfig
+metadata:
+  name: uds-core-runtime
+  description: "UDS Core Runtime"
+  url: "https://github.com/defenseunicorns/uds-runtime"
+
+components:
+  - name: uds-runtime
```
Comment (rjferguson21) [inline src/runtime/zarf.yaml:8]: Could/Should we be importing this from the published runtime package?

Comment (mjnagel) [reply src/runtime/zarf.yaml:8]: Unfortunately that's only possible with skeleton I believe - and I'm not sure runtime is publishing/should be publishing a skeleton package? I think that pattern is generally being discouraged?

Comment (UncleGedd) [reply src/runtime/zarf.yaml:8]: I think you can only import remote components as skeleton packages, which we don't currently publish

Comment (UncleGedd) [reply src/runtime/zarf.yaml:8]: I should refresh more lol. Just deleted my comment saying ^^. I was also under the impression 💀 packages are discouraged

File: src/runtime/zarf.yaml (hunk 1 above)
Comment (UncleGedd) [reply src/runtime/zarf.yaml:15]: Made an issue to address: https://github.com/defenseunicorns/uds-runtime/issues/355

**Review by rjferguson21 — APPROVED** [review]

**Pull request merged**